resources/aws/securitygroup.go

```diff
@@ -15,8 +19,47 @@ type SecurityGroup struct {
 	AWSEntity
 }
 
+func (s *SecurityGroup) existingSecurityGroupID() (string, error) {
+	securityGroups, err := s.Clients.EC2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name: aws.String("description"),
+				Values: []*string{
+					aws.String(s.Description),
+				},
+			},
+			&ec2.Filter{
+				Name: aws.String("group-name"),
+				Values: []*string{
+					aws.String(s.GroupName),
+				},
+			},
+		},
+	})
+	if err != nil {
+		return "", microerror.MaskAny(err)
+	}
+
+	for _, securityGroup := range securityGroups.SecurityGroups {
```

Comment:
I would check if `securityGroups` is empty and return an error, to be consistent with other places in the code: we end with the happy path, and return errors in `if` clauses.

I think it also makes the code more explicit. WDYT?